cmd/ttn-lw-cli/commands/gateways.go

```diff
@@ -48,6 +48,53 @@ func gatewayIDFlags() *pflag.FlagSet {
 
 var errNoGatewayID = errors.DefineInvalidArgument("no_gateway_id", "no gateway ID set")
 
+func gatewayClaimAuthCodeValidityFlags() *pflag.FlagSet {
+	flagSet := &pflag.FlagSet{}
+	flagSet.String("claim-authentication-code.valid-from", "", "(YYYY-MM-DDTHH:MM:SSZ)")
+	flagSet.String("claim-authentication-code.valid-to", "", "(YYYY-MM-DDTHH:MM:SSZ)")
+	return flagSet
+}
+
+func getgatewayClaimAuthCodeSecret(flagSet *pflag.FlagSet, gateway ttnpb.Gateway) (*ttnpb.Secret, error) {
+	if gateway.ClaimAuthenticationCode == nil {
+		return nil, nil
+	}
+
+	gatewayClaimAuthCode := ttnpb.GatewayClaimAuthenticationCode{
+		Value: gateway.ClaimAuthenticationCode.Value,
+	}
+	validFromString, err := flagSet.GetString("claim-authentication-code.valid-from")
+	if err != nil {
+		return nil, err
+	}
+	if validFromString != "" {
+		validFrom, err := time.Parse(time.RFC3339, validFromString)
+		if err != nil {
+			return nil, err
+		}
+		gatewayClaimAuthCode.ValidFrom = &validFrom
+	}
+	validToString, err := flagSet.GetString("claim-authentication-code.valid-to")
+	if err != nil {
+		return nil, err
+	}
+	if validToString != "" {
+		validTo, err := time.Parse(time.RFC3339, validToString)
+		if err != nil {
+			return nil, err
+		}
+		gatewayClaimAuthCode.ValidTo = &validTo
+	}
+	ret, err := gatewayClaimAuthCode.Marshal()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I really don't like this. Only the code value is secret. The timestamps are not.